lib/puppet/network/authconfig.rb

```diff
@@ -89,4 +89,28 @@ def initialize(rights=nil)
       insert_default_acl
     end
   end
+
+  class Network::AuthConfig
+    @@authprovider_class = nil
+
+    def self.authprovider_class=(klass)
+      @@authprovider_class = klass
+    end
+
+    def self.authprovider_class
+      @@authprovider_class || Puppet::Network::DefaultAuthProvider
+    end
+
+    def initialize(rights=nil)
+      if rights.nil?
```

Comment:
Discussed further with @camlow325 - let's just pass down `rights` regardless of nil or not and let the implementations of authprovider_class deal with nil accordingly.